src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

```diff
@@ -542,7 +543,7 @@ public boolean gatherIfInTileEntity(final TileEntity entity, final ItemStack is)
             return false;
         }
 
-        if (InventoryUtils.hasEnoughInProvider(entity, is, is.getCount()))
+        if (entity instanceof TileEntityRack && ((TileEntityRack) entity).getCount(new ItemStorage(is)) >= is.getCount())
```

Comment:
This does not seem to work when a courier is gathering crafted items from buildings. TileEntityColonyBuilding is inherited from TileEntityRack, but the getCount method returns 0 when querying for the itemstack/storage